example/Main.hs

```diff
@@ -142,26 +129,21 @@ nextLspReqId = do
 -- | The single point that all events flow through, allowing management of state
 -- to stitch replies and requests together from the two asynchronous sides: lsp
 -- server and backend compiler
-reactor :: ReactorState -> TChan ReactorInput -> IO ()
-reactor st inp = do
+reactor :: Core.LspFuncs -> ReactorState -> TChan ReactorInput -> IO ()
+reactor lf st inp = do
   liftIO $ U.logs $ "reactor:entered"
-  flip evalStateT st $ forever $ do
+  flip evalStateT st $ flip runReaderT lf $ forever $ do
     inval <- liftIO $ atomically $ readTChan inp
     case inval of
-      -- This will be the first message received from the client, and can be
-      -- used to tailor processing according to the given client capabilities.
-      InitializeCallBack lf@(Core.LspFuncs capabilities _sf _vf _dpf) -> do
```

Comment:
Does the server get a handler call with this information, so that it can set its internal processing according to the client capablities, and advertise its own?